src/coreclr/jit/lower.cpp

```diff
@@ -2020,51 +2023,77 @@ GenTree* Lowering::LowerCallMemcmp(GenTreeCall* call)
                     GenTree* rArgClone = comp->gtNewLclvNode(rArgUse.ReplaceWithLclVar(comp), genActualType(rArg));
                     BlockRange().InsertBefore(call, lArgClone, rArgClone);
 
-                    // We're going to emit something like the following:
-                    //
-                    // bool result = ((*(int*)leftArg ^ *(int*)rightArg) |
-                    //                (*(int*)(leftArg + 1) ^ *((int*)(rightArg + 1)))) == 0;
-                    //
-                    // ^ in the given example we unroll for length=5
-                    //
-                    // In IR:
-                    //
-                    // *  EQ        int
-                    // +--*  OR        int
-                    // |  +--*  XOR       int
-                    // |  |  +--*  IND       int
-                    // |  |  |  \--*  LCL_VAR   byref  V1
-                    // |  |  \--*  IND       int
-                    // |  |     \--*  LCL_VAR   byref  V2
-                    // |  \--*  XOR       int
-                    // |     +--*  IND       int
-                    // |     |  \--*  ADD       byref
-                    // |     |     +--*  LCL_VAR   byref  V1
-                    // |     |     \--*  CNS_INT   int    1
-                    // |     \--*  IND       int
-                    // |        \--*  ADD       byref
-                    // |           +--*  LCL_VAR   byref  V2
-                    // |           \--*  CNS_INT   int    1
-                    // \--*  CNS_INT   int    0
-                    //
+                    *next = lArgClone;
+
                     GenTree* l1Indir   = comp->gtNewIndir(loadType, lArgUse.Def());
                     GenTree* r1Indir   = comp->gtNewIndir(loadType, rArgUse.Def());
-                    GenTree* lXor      = newBinaryOp(comp, GT_XOR, actualLoadType, l1Indir, r1Indir);
                     GenTree* l2Offs    = comp->gtNewIconNode(cnsSize - loadWidth, TYP_I_IMPL);
                     GenTree* l2AddOffs = newBinaryOp(comp, GT_ADD, lArg->TypeGet(), lArgClone, l2Offs);
                     GenTree* l2Indir   = comp->gtNewIndir(loadType, l2AddOffs);
-                    GenTree* r2Offs    = comp->gtCloneExpr(l2Offs); // offset is the same
+                    GenTree* r2Offs    = comp->gtNewIconNode(cnsSize - loadWidth, TYP_I_IMPL);
```

Comment:
motivation? 🙂

Reply:
No strong motivation... just that we were already creating all the other constants from scratch so it looked a bit out of place to me to clone one lone one. I can revert it if you prefer

Reply:
I am fine with this, was just wondering if you were improving TP impact or something like that